src/main/java/org/generationcp/breeding/manager/listimport/listeners/GidLinkClickListener.java

```diff
@@ -95,45 +91,32 @@ private void openDetailsWindow(final Component component) {
 			mainWindow = component.getApplication().getWindow(GidLinkClickListener.GERMPLASM_IMPORT_WINDOW_NAME);
 		}
 
-		Tool tool = null;
-		try {
-			tool = this.workbenchDataManager.getToolWithName(ToolName.GERMPLASM_BROWSER.toString());
-		} catch (MiddlewareQueryException qe) {
-			GidLinkClickListener.LOG.error("QueryException", qe);
-		}
-		String addtlParams = Util.getAdditionalParams(this.workbenchDataManager);
-
-		ExternalResource germplasmBrowserLink = null;
-		if (tool == null) {
-			germplasmBrowserLink =
-					new ExternalResource(WorkbenchAppPathResolver.getFullWebAddress(DefaultGermplasmStudyBrowserPath.GERMPLASM_BROWSER_LINK
-							+ this.gid, "?restartApplication" + addtlParams));
-		} else {
-			germplasmBrowserLink =
-					new ExternalResource(WorkbenchAppPathResolver.getWorkbenchAppPath(tool, this.gid, "?restartApplication" + addtlParams));
-		}
+		final ExternalResource germplasmDetailsLink = new ExternalResource(
```

Comment:
Can we extract this code to utility (passing gid, and contextUtil) so we don't form in multiple places in Vaadin:
```
final ExternalResource germplasmDetailsLink = new ExternalResource(
			WorkbenchAppPathResolver.getFullWebAddress(DefaultGermplasmStudyBrowserPath.GERMPLASM_DETAILS_LINK + this.gid + "?cropName="
				+ this.contextUtil.getProjectInContext().getCropType().getCropName() + "&programUUID=" + this.contextUtil
				.getCurrentProgramUUID()) + "&modal");
```

Reply:
This is done.